tests/test_client.py

```diff
@@ -1,219 +1,109 @@
-
-        assert "api_key" in message
-        assert "client_id" not in message
-
-    def test_initialize_sso_missing_client_id(self, set_api_key):
-        with pytest.raises(ConfigurationException) as ex:
-            client.sso
-
-        message = str(ex)
-
-        assert "client_id" in message
-        assert "api_key" not in message
-
-    def test_initialize_sso_missing_api_key_and_client_id(self):
-        with pytest.raises(ConfigurationException) as ex:
-            client.sso
-
-        message = str(ex)
-
-        assert all(
-            setting in message
-            for setting in (
-                "api_key",
-                "client_id",
-            )
+class TestClient:
+    @pytest.fixture
+    def default_client(self):
+        return WorkOSClient(
+            api_key="sk_test", client_id="client_b27needthisforssotemxo"
         )
 
-    def test_initialize_directory_sync_missing_api_key(self):
-        with pytest.raises(ConfigurationException) as ex:
-            client.directory_sync
-
-        message = str(ex)
-
-        assert "api_key" in message
-
-    def test_initialize_events_missing_api_key(self):
-        with pytest.raises(ConfigurationException) as ex:
-            client.events
-
-        message = str(ex)
+    def test_client_without_api_key(self):
+        with pytest.raises(ValueError) as error:
+            WorkOSClient(client_id="client_b27needthisforssotemxo")
 
-        assert "api_key" in message
-
-    def test_initialize_mfa_missing_api_key(self):
-        with pytest.raises(ConfigurationException) as ex:
-            client.mfa
-
-        message = str(ex)
-
-        assert "api_key" in message
-
-    def test_initialize_organizations_missing_api_key(self):
-        with pytest.raises(ConfigurationException) as ex:
-            client.organizations
-
-        message = str(ex)
-
-        assert "api_key" in message
-
-    def test_initialize_passwordless_missing_api_key(self):
-        with pytest.raises(ConfigurationException) as ex:
-            client.passwordless
-
-        message = str(ex)
-
-        assert "api_key" in message
-
-    def test_initialize_portal_missing_api_key(self):
-        with pytest.raises(ConfigurationException) as ex:
-            client.portal
-
-        message = str(ex)
-
-        assert "api_key" in message
-
-    def test_initialize_user_management_missing_client_id(self, set_api_key):
-        with pytest.raises(ConfigurationException) as ex:
-            client.user_management
-
-        message = str(ex)
-
-        assert "client_id" in message
+        assert (
+            "WorkOS API key must be provided when instantiating the client or via the WORKOS_API_KEY environment variable."
+            == str(error.value)
+        )
 
-    def test_initialize_user_management_missing_api_key(self, set_client_id):
-        with pytest.raises(ConfigurationException) as ex:
-            client.user_management
+    def test_client_without_client_id(self):
```

Comment:
Talking through why we need this fixture at all: the SDK has no notion of which API methods need a client_id, so this is essentially emulating an error response from the API?

Reply:
Yep, that's correct. If we decide we shouldn't required a client ID for every instantiation, we'll want to rework.